### DISCUSSION TITLE
Iterating on top level floorplan with macro pin constraints

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#2927] [General]
I would like to iterate on the floorplan of the top level of our design. In doing so, I want to change the pin constraints for the macros used at the top level.

However, ```make generate_abstract``` goes all the way to detailed routing before producing a .gds, .lef and .lib file. This takes a long time.

Is there a way to run generate_abstract after IO pin placement on the macro without going through detailed route?

**Comment by maliberty** [comment]
The flow itself doesn't have a mechanism but the command should work at that point (though it hasn't been tested).  You could add a flow option if needed.

**Comment by oharboe** [comment]
Solution implemented: `make skip_cts skip_route generate_abstract`
